### PR TITLE
feat: Use batch json rpc when using ethers

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

I wanted to test this and see if it can improve performance. Opening a draft PR for now. 

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/src/fulfillment/dispatcher.ts [hunk 1]
```diff
@@ -16,10 +16,14 @@ interface KeyToGrant {
   expiration?: number
 }
 export default class Dispatcher {
-  async getPurchaser(network: number) {
-    const provider = new ethers.providers.JsonRpcProvider(
+  providerForNetwork(network: number) {
+    const provider = new ethers.providers.JsonRpcBatchProvider(
```
Comment (julien51) [inline locksmith/src/fulfillment/dispatcher.ts:20]: The dispatcher sends transactions, so I am not sure batching makes much sense here!

Comment (searchableguy) [reply locksmith/src/fulfillment/dispatcher.ts:20]: Yeah, I replaced the use here for consistency since it would affect the behavior either way.

Comment (julien51) [reply locksmith/src/fulfillment/dispatcher.ts:20]: What do you think of _not_ changing it for now then so we can test more thoroughly and avoid breaking things in unexpected ways.

**Comment by searchableguy** [comment]
Rebased the PR.

**Review by clemsos — COMMENTED** [review]
LGTM

File: smart-contracts/scripts/multisig/_helpers.js [hunk 2]
```diff
@@ -1,50 +1,50 @@
...
+const multisigOldABI = require("../../test/helpers/ABIs/multisig.json");
 
 // get the correct provider if chainId is specified
 const getProvider = async (chainId) => {
-  let provider
+  let provider;
   if (chainId) {
-    const { publicProvider } = networks[chainId]
-    provider = new ethers.providers.JsonRpcProvider(publicProvider)
+    const { publicProvider } = networks[chainId];
```
[the remark this answers is not in the document]
Comment (searchableguy) [reply smart-contracts/scripts/multisig/_helpers.js:9]: Just changed it for consistency.

**Review by julien51 — APPROVED** [review]
LGTM

**Pull request merged**